ipc/grpc/client/src/main/java/org/opennms/poc/ignite/grpc/client/GrpcClient.java

```diff
@@ -0,0 +1,36 @@
+package org.opennms.poc.ignite.grpc.client;
+
+import io.grpc.ConnectivityState;
+import java.io.Closeable;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.opennms.cloud.grpc.minion.CloudToMinionMessage;
+import org.opennms.cloud.grpc.minion.MinionToCloudMessage;
+import org.opennms.cloud.grpc.minion.RpcRequest;
+import org.opennms.cloud.grpc.minion.RpcResponse;
+
+public interface GrpcClient extends AutoCloseable {
+
+  // CloudToMinionRPC, async
+  Session onCall(Predicate<RpcRequest> request, Function<RpcRequest, CompletableFuture<RpcResponse>> response);
```

Comment:
Why is the response a method param? Shouldn't it be the return value? Side effects BAD!

Reply:
This signature covers a incoming rpc request from server, when server requests an operation on the minion side.
Signature is: `Predicate<Request>, Function<Request, Response>`. It allows a module register itself as an incoming rpc request listener, hence predicate as first and function as a second argument. Response is created by listener itself, client is responsible only for forwarding method call. Client itself does not modify response contents.
Maybe later on we can modify that part and create `RpcResponse` automatically within client to flatten function signature.